--- a/constants.py	2019-04-25 13:54:51.740610037 -0700
+++ b/constants.py	2019-04-25 13:55:14.996430304 -0700
@@ -15,12 +15,7 @@
 """Plugins Setuptools entry point before rename."""
 
 CLI_DEFAULTS = dict(
-    config_files=[
-        os.path.join(misc.get_default_folder('config'), 'cli.ini'),
-        # http://freedesktop.org/wiki/Software/xdg-user-dirs/
-        os.path.join(os.environ.get("XDG_CONFIG_HOME", "~/.config"),
-                     "letsencrypt", "cli.ini"),
-    ],
+    config_files=[],
 
     # Main parser
     verbose_count=-int(logging.INFO / 10),
