--- a/_internal/constants.py
+++ b/_internal/constants.py
@@ -16,12 +16,7 @@ OLD_SETUPTOOLS_PLUGINS_ENTRY_POINT = "letsencrypt.plugins"
 """Plugins Setuptools entry point before rename."""
 
 CLI_DEFAULTS: Dict[str, Any] = dict(
-    config_files=[
-        os.path.join(misc.get_default_folder('config'), 'cli.ini'),
-        # https://freedesktop.org/wiki/Software/xdg-user-dirs/
-        os.path.join(os.environ.get("XDG_CONFIG_HOME", "~/.config"),
-                     "letsencrypt", "cli.ini"),
-    ],
+    config_files=[],
 
     # Main parser
     verbose_count=0,
