--- a/_internal/constants.py      2021-10-13 11:27:41.539254415 +0000
+++ b/_internal/constants.py      2021-10-13 11:37:41.203246935 +0000
@@ -15,10 +15,6 @@
 
 CLI_DEFAULTS = dict(
     config_files=[
-        os.path.join(misc.get_default_folder('config'), 'cli.ini'),
-        # https://freedesktop.org/wiki/Software/xdg-user-dirs/
-        os.path.join(os.environ.get("XDG_CONFIG_HOME", "~/.config"),
-                     "letsencrypt", "cli.ini"),
     ],
 
     # Main parser
