--- a/compat/filesystem.py	2026-03-16 22:05:19.000000000 +0000
+++ b/compat/filesystem.py	2026-03-16 22:05:38.000000000 +0000
@@ -69,7 +69,6 @@
         stats = os.stat(src)
         user_id = stats.st_uid if copy_user else -1
         group_id = stats.st_gid if copy_group else -1
-        os.chown(dst, user_id, group_id)
     elif copy_user:
         # There is no group handling in Windows
         _copy_win_ownership(src, dst)
