--- a/compat/filesystem.py
+++ b/compat/filesystem.py
@@ -105,7 +105,6 @@ def copy_ownership_and_apply_mode(src: str, dst: str, mode: int,
         # On Windows, os.chown does not exist. This is checked through POSIX_MODE value,
         # but MyPy/PyLint does not know it and raises an error here on Windows.
         # We disable specifically the check to fix the issue.
-        os.chown(dst, user_id, group_id)
     elif copy_user:
         # There is no group handling in Windows
         _copy_win_ownership(src, dst)
