--- a/compat/filesystem.py	2026-05-07 21:43:53.000000000 +0000
+++ b/compat/filesystem.py	2026-05-07 21:44:04.000000000 +0000
@@ -102,7 +102,6 @@
         # On Windows, os.chown does not exist. This is checked through POSIX_MODE value,
         # but MyPy/PyLint does not know it and raises an error here on Windows.
         # We disable specifically the check to fix the issue.
-        os.chown(dst, user_id, group_id)
     elif copy_user:
         # There is no group handling in Windows
         _copy_win_ownership(src, dst)
